AlgorithmFactory/Python/Wrappers/AlgorithmPythonWrapper.cs

```diff
@@ -558,7 +558,20 @@ public void OnData(Slice slice)
         /// Used to send data updates to algorithm framework models
         /// </summary>
         /// <param name="slice">The current data slice</param>
-        public void OnFrameworkData(Slice slice) => _baseAlgorithm.OnFrameworkData(slice);
+        public void OnFrameworkData(Slice slice)
+        {
+            if (SubscriptionManager.HasCustomData)
+            {
+                using (Py.GIL())
+                {
+                    _algorithm.OnFrameworkData(new PythonSlice(slice));
+                }
+            }
+            else
+            {
+                _baseAlgorithm.OnFrameworkData(slice);
```

Comment:
Wondering why one is `_algorithm.OnFrameworkData` and the other `_baseAlgorithm.OnFrameworkData`? Might be cleaner if it were 1 liner `SubscriptionManager.HasCustomData ? new PythonSlice(slice) : slice`

Reply:
In my version 1.0, I implemented it line `OnData`.
However, if we don't have custom data, we don't need to use the Py.GIL lock.